src/callbacks_step/visualization.jl

```diff
@@ -0,0 +1,223 @@
+  end
+
+  DiscreteCallback(visualization_callback, visualization_callback, # the first one is the condition, the second the affect!
+                   save_positions=(false,false),
+                   initialize=initialize!)
+end
+
+
+function initialize!(cb::DiscreteCallback{Condition,Affect!}, u, t, integrator) where {Condition, Affect!<:VisualizationCallback}
+  visualization_callback = cb.affect!
+
+  visualization_callback(integrator)
+
+  return nothing
+end
+
+
+# this method is called to determine whether the callback should be activated
+function (visualization_callback::VisualizationCallback)(u, t, integrator)
+  @unpack interval = visualization_callback
+
+  return interval > 0 && ((integrator.iter % interval == 0) || isfinished(integrator))
+end
+
+
+# this method is called when the callback is activated
+function (visualization_callback::VisualizationCallback)(integrator)
+  u_ode = integrator.u
+  semi = integrator.p
+  @unpack plot_arguments, solution_variables, variable_names, show_mesh, plot_data_creator, plot_creator = visualization_callback
+
+  # Extract plot data
+  plot_data = plot_data_creator(u_ode, semi, solution_variables=solution_variables)
+
+  # If variable names were not specified, plot everything
+  if isempty(variable_names)
+    variable_names = String[keys(plot_data)...]
+  end
+
+  # Create plot
+  plot_creator(plot_data, variable_names;
+               show_mesh=show_mesh, plot_arguments=plot_arguments,
+               time=integrator.t, timestep=integrator.iter)
+
+  # avoid re-evaluating possible FSAL stages
+  u_modified!(integrator, false)
+  return nothing
+end
+
+
+"""
+    show_plot(plot_data, variable_names;
+              show_mesh=true, plot_arguments=Dict{Symbol,Any}(),
+              time=nothing, timestep=nothing)
+
+Visualize the plot data object provided in `plot_data` and display result, plotting only the
+variables in `variable_names` and, optionally, the mesh (if `show_mesh` is `true`).  Additionally,
+`plot_arguments` will be unpacked and passed as keyword arguments to the `Plots.plot` command.
+
+This function is the default `plot_creator` argument for the [`VisualizationCallback`](@ref).
+`time` and `timestep` are currently unused by this function.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+
+See also: [`VisualizationCallback`](@ref), [`save_plot`](@ref)
+"""
+function show_plot(plot_data, variable_names;
+                   show_mesh=true, plot_arguments=Dict{Symbol,Any}(),
+                   time=nothing, timestep=nothing)
+  # Gather subplots
+  plots = []
+  for v in variable_names
+    push!(plots, plot(plot_data[v]; plot_arguments...))
+  end
+  if show_mesh
+    push!(plots, plot(getmesh(plot_data); plot_arguments...))
+  end
+
+  # Determine layout
+  cols = ceil(Int, sqrt(length(plots)))
+  rows = ceil(Int, length(plots)/cols)
```

Comment:
```suggestion
  rows = length(plots) ÷ cols
```
This performs integer division (which you want to use here, I guess).

Reply:
Ah, I mixed up the rounding directions. I Fixed my post above.

Reply:
Thanks for the fix! And I love the ASCII-only solution ;-)

Reply:
Fixed in 3bea77ad.